modules/single-node-asg/variables.tf

```diff
@@ -115,3 +79,15 @@ variable "load_balancers" {
   description = "The list of load balancers names to pass to the ASG module"
   type        = list(string)
 }
+
+variable "data_volumes" {
+  type = list(map(any))
+  #    type = string,
+  #    iops = number,
+  #    size = number,
+  #    encrypted = bool,
+  #    kms_key_id = string,
+  #    snapshot_id = string,
+  #    name = string,
+  #    device = string
+}
```

Comment:
Did the map spec fail, or why is it commented out here?

Reply:
I designed this as object, but it did not work as expected with `merge`, so converted to map.